android/libraries/rib-base/src/main/java/com/badoo/ribs/android/PermissionRequester.kt

```diff
@@ -0,0 +1,45 @@
+package com.badoo.ribs.android
+
+import com.badoo.ribs.core.Identifiable
+import io.reactivex.Observable
+
+interface PermissionRequester {
+
+    fun checkPermissions(
+        client: Identifiable,
+        permissions: Array<String>
+    ) : CheckPermissionsResult
+
+    fun requestPermissions(
+        client: Identifiable,
+        requestCode: Int,
+        permissions: Array<String>
+    )
+
+    fun events(
+        client: Identifiable,
+        requestCode: Int
+    ): Observable<RequestPermissionsEvent>
+
+    data class CheckPermissionsResult(
+        val granted: List<String>,
+        val notGranted: List<String>,
+        val shouldShowRationale: List<String>
+    ) {
+        fun allGranted() : Boolean =
```

Comment:
Can we replace it to val, not fun? for me looks like a val as other fields

Reply:
In my mind it's a fun, but I'm open to hear others as well.

Reply:
Agree with val here.
Could be an extension as well.

Reply:
Done.

Reply:
`val` from my point of view.